packages/sdk/src/transactions/PSBTBuilder.ts

```diff
@@ -317,7 +286,11 @@ export class PSBTBuilder extends FeeEstimator {
     if (!this.autoAdjustment && !address) return
 
     const amountToRequest =
-      amount && amount > 0 ? amount : this.changeAmount < 0 ? this.changeAmount * -1 : this.outputAmount
+      amount && amount > 0
+        ? amount
+        : this.changeAmount < 0
+        ? this.changeAmount * -1
+        : this.outputAmount - this.getRetrievedUTXOsValue()
```

Comment:
what happens when `this.outputAmount - this.getRetrievedUTXOsValue())` is negative?  then we should not call getSpendables anymore.

Reply:
Good catch!

Reply:
fixed; https://github.com/sadoprotocol/ordit-sdk/pull/81/commits/d4d8955b17866421fcf7fbffd91a491449f35c86